src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -2109,25 +2119,68 @@ public boolean next() throws SQLException {
           String pk_name = showObjectResultSet.getString(7);
 
           // Post filter based on the input
-          if ((catalogIn == null || catalogIn.equals(table_cat))
-              && (compiledSchemaPattern == null
-                  || compiledSchemaPattern.equals(table_schem)
-                  || compiledSchemaPattern.matcher(table_schem).matches())
-              && (compiledTablePattern == null
-                  || compiledTablePattern.equals(table_name)
-                  || compiledTablePattern.matcher(table_name).matches())) {
-            nextRow[0] = table_cat;
-            nextRow[1] = table_schem;
-            nextRow[2] = table_name;
-            nextRow[3] = column_name;
-            nextRow[4] = key_seq;
-            nextRow[5] = pk_name;
-            return true;
+          if (isPatternMatchingEnabled) {
+            if (isPrimaryKeyPatternSearch(
+                table_cat, table_schem, table_name, column_name, key_seq, pk_name)) return true;
+          } else {
+            if (isPrimaryKeyExactSearch(
+                table_cat, table_schem, table_name, column_name, key_seq, pk_name)) return true;
           }
         }
         close();
         return false;
       }
+
+      private boolean isPrimaryKeyExactSearch(
+          String table_cat,
+          String table_schem,
+          String table_name,
+          String column_name,
+          int key_seq,
+          String pk_name) {
+        if ((catalogIn == null || catalogIn.equals(table_cat))
+            && (schemaIn == null || schemaIn.equals(table_schem))
+            && (tableIn == null || tableIn.equals(table_name))) {
+          createPrimaryKeyRow(table_cat, table_schem, table_name, column_name, key_seq, pk_name);
```

Comment:
let's run this method outside this method based on result - the method name does not show that such modification happen inside and indeed it should be done outside

Reply:
the method `createPrimaryKeyRow ` is fine - let's just call it outside of `is...` methods

Reply:
Oh I see. I misunderstood. I updated it now so it should be correct.